### PR TITLE
Dev popup, triggered by hotkey, for populating useful data

### PR DESCRIPTION
**Pull request by sconover**
- ctl+shift+d causes the dev dialog to appear
- Buttons in the dialog cause dev actions to execute,
  such as simulating the drop of a python file
- Each time a dev action is clicked, it's added to the hash
  part of the url

extract top level action adapter functions. dev module - initial.
ctl+shift+d to open dialog, esc to close
file drop example
auto-run dev actions when specified in location hash
update location hash when dev actions change
extract set location hash
convertLocationHashToUrlFragment: test
more tests

### CODE REVIEW

File: src/ts/app/console/src/reducer.ts [hunk 1]
```diff
@@ -10,7 +10,7 @@ export interface ReducerInputs {
 }
 
 export function createReducer(inputs: ReducerInputs): ImprovedLoopReducer<Subtree, HandledAction, any> {
-    return function reduce(previous: Subtree, action: HandledAction): Subtree | [Subtree, HandledAction] {
+    return function reduce(previous: Subtree, action: HandledAction): Subtree | [Subtree, any] {
```
Comment (daicoden) [inline src/ts/app/console/src/reducer.ts:13]: why? does't this break the typing?

File: src/ts/main/src/install-drag-drop-action-adapters.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        // - handle multiple files
+        // - put in place other defensiveness like checking files sizes and such
+        //
+        // This functionality should be considered prototype/demonstration-quality for now.
+
+        const fileList = e.dataTransfer!.files
+        for (let i = 0; i < fileList.length; i++) {
+            const reader = new FileReader()
+            reader.addEventListener('load', (event) => {
+                dispatch({
```
Comment (daicoden) [inline src/ts/main/src/install-drag-drop-action-adapters.ts:26]: do you need to unsubscribe the eventListener in here?

File: src/ts/app/console/src/reducer.ts (hunk 1 above)
Comment (sconover) [reply src/ts/app/console/src/reducer.ts:13]: There's no `DispatchedAction` yet (which is the type that's _returned_ from the reducer), so it needs to be `any`.

File: src/ts/main/src/install-drag-drop-action-adapters.ts (hunk 2 above)
Comment (sconover) [reply src/ts/main/src/install-drag-drop-action-adapters.ts:26]: Two things:

a) This code is very "practical", i.e. kind of a hack/PoC. So there's that.
b) This code should run exactly once, on single-page-app "boot". So I don't think there's any need for an unsubscribe, if that holds true.

**Pull request merged**